laws/shared/src/main/scala/scalaz/laws/HomomorphismLaws.scala

```diff
@@ -0,0 +1,191 @@
+package scalaz
+package laws
+
+import scala.{ Nothing, Unit }
+
+import data._, tc._, Scalaz._
+
+object HomomorphismLaws {
+  // To be a monoid homomorphism, a function must satisfy both `monoidIdentity`
+  // and `semigroupAppend`.
+  def monoidIdentity[A, B, T](f: A => B)(assert: (B, B) => T)(implicit A: Monoid[A], B: Monoid[B]): T =
```

Comment:
If this is testing homomorphisms, then shouldn't it be testing in both directions?  This only verifies half of the homomorphism, unless I'm misunderstanding something.

(Same question applies to other assertions in this file)

Reply:
I don't understand; homomorphisms go in one direction.

Reply:
Though that reminds me, it would be nice to have isomorphism laws too.

Reply:
Oh yeah my bad. I got mixed up. Pretend that I was telling you to write `Iso` laws.